src/main/kotlin/se/uulm/snowballr/backend/repository/association/CitationTableRepo.kt

```diff
@@ -32,8 +32,8 @@ class CitationTableRepo(
 ) : ICitationTableRepo {
     override suspend fun getBackwardsReferencedPaperIdsOfPaperById(id: UUID): List<UUID> = db.query {
         CitationTable
-            .select(CitationTable.paperId)
-            .where { CitationTable.citedPaperId eq id }
+            .select(CitationTable.citedPaperId)
+            .where { CitationTable.paperId eq id }
             .map { it[CitationTable.paperId].value }
```

Comment:
```suggestion
            .map { it[CitationTable.citedPaperId].value }
```
One thing left: my suggestion was not completely correct. You have to map the correct value, as paperId doesn't exist anymore after selecting the citedPaperId.

And you have to adapt the tests in the `CitationTableRepoTest`.

Reply:
Sorry, because of lack of time I didn't run the tests last time. Sorry for your extra effort. Should be fixed now.